react/week3/src/TodoList.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import TodoItem from './TodoItem';
+import { v4 as uuidv4 } from 'uuid';
+import Form from './Form'
+
+
+
+class TodoList extends React.Component {
+  state = { todos: [ ] }
+
+addTodo = (todo, dueDate) => {
+    const newTodo = {id: uuidv4(), todo, dueDate, complete: false}
+    this.setState({
+        todos: [
+            ...this.state.todos,
+            newTodo 
+        ]
+    })
+}; 
+
+deleteTodo = (currentId) => {
+    this.setState({
+        todos: this.state.todos.filter((_todo, id) => currentId !== id)
+    });
+}
+
+toggleTodo = (currentId) => {
+    this.setState({
+        todos: this.state.todos.map((todo, id) => {
+            if(currentId === id) {
+                return {
+                    ...todo, 
+                    complete: !todo.complete
+                };
+            } else {
+                return todo
+            }
+        })
+    })
+};
+
+// editTodo = (currentId) => {
+//     this.setState({
+//         todos: 
+//     })
+// }
+
+render() {
+    return (
+        <div>
+            <ul>
+                {this.state.todos.map((todo, id, dueDate) => (
```

Comment:
It looks like what you might be trying to do here is destructure the element you're mapping over. If that's what you wanted, this would be written like this: 
```suggestion
                {this.state.todos.map(({ todo, id, dueDate }) => (
```

Reply:
I feel like you may have been dealing with some unexpected bugs and my guess is this is the cause.

Reply:
I don't have issues with the render, it's been working fine. I tried to implemented this but it doesn't seem to work, so I omitted.  With type of issues can I find in this case?